### ISSUE TITLE
extract and loop models

### ISSUE THREAD

**Issue by Josh-Lee1** [Josh-Lee1/eBird-Fire-Index#9]
Hey @wcornwell and @coreytcallaghan, 
I've played a bit with writing a function to extract the coefficients and se for each species model. 
Not sure if I've got the most elegant method but I'm still not 100% on the syntax of function building. 
This is the code I have now.  I got each bit to work so I think its just the syntax of the function that isn't working.
Am i missing anything obvious? 
Thanks

  extractingcoeffse_function <- function(file_name) {
    data <- readRDS(file_name)
    datsum<- summary(data)
    coeff<- dat$coefficients
    se<- datsum$se
    wgoods<- data.frame(coeff, se) 
    
    saveRDS(wgoods, file = paste0("models_extracted/", gsub("model_objects/", "",file_name), ".RDS"))
  }

**Comment by coreytcallaghan** [comment]
It looks good!

I would try to pull out a bit more info. Number of observations used in the
model (or will this be the same for every species?), deviance explained of
the model, and anything else that might come in handy...

Looks solid though. Just need to apply/loop it now!

**Comment by Josh-Lee1** [comment]
Would it be better to store the data as a df or a list?

**Comment by coreytcallaghan** [comment]
Dataframe!

**Comment by wcornwell** [comment]
maybe something like this at the end of the function:

```{r}
return(tibble(species=file_name, coeff=coeff[2], se=se[2]))
```
so it just returns a data frame with one row. Then it will play nicely with the fancy loop in `map_df`

```{r}
  bird_response_df <- map_df(files, extractingcoeffse_function)
```

**Comment by Josh-Lee1** [comment]
Ok I think I have written things that are roughly what we want. I think I still don't fully understand when to pass what into where?? 
This might be why I cant get things to run. I would appreciate some help finding where my problems are. 
I pushed the changes to "loop_for_models.R".

**Comment by Josh-Lee1** [comment]
Also Number of observations will be different but Number of checklists will be the same right?

**Comment by coreytcallaghan** [comment]
Hey @Josh-Lee1.

Thanks for your patience.

I just messed with this a bit. Made it a little bit simpler. Here, I wouldn't write out the files again as it is relatively low overhead to calculate the things and pull them into one df.... so I would just return the summary df!

Also, I got rid of the smooth estimates and such as I don't think they are necessary.

Now, it should work fine. A test (for the first five species) works for me: ```bird_response_df <- map_df(all_files[1:5], extractingcoeffse_function)```.

Should be able to let it run for all of them now. 

Let me know if this is working!
